### PR TITLE
ARM: Support for Broadcom IPROC architecture and Accton AS4610-54 (Broadcom Helix4 SOC) Board

### PR DESCRIPTION
**Pull request by vidyaravipati**
busybox sys_eeprom: Support reading/writing the "eeprom" exported in sysfs file
 u-boot: Support for U-Boot version 2012.10 in ONIE core
 arm: iproc: ONIE Support for Broadcom IPROC architecture
arm: accton AS4610-54: ONIE support for accton as4610-54 board

### CODE REVIEW

File: machine/accton/accton_as4610_54/INSTALL [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+type ``"make MACHINEROOT=../machine/accton MACHINE=accton_as4610_54 all"``.
+For example::
+
+  $ cd build-config
+  $ make -j4 MACHINEROOT=../machine/accton MACHINE=accton_as4610_54 all
+
+When complete, the ONIE binaries are located in
+``build/images``::
+
+  -rw-r--r-- 3604480 May  7 11:38 onie-accton_as4610_54-r0.bin
```
Comment (lakesh-sharma) [inline machine/accton/accton_as4610_54/INSTALL]: Can we list the other binaries built as well. Just for info purpose.

File: machine/accton/accton_as4610_54/machine.make [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+# Translate hardware revision to ONIE hardware revision
+ifeq ($(VENDOR_REV),0)
+  MACHINE_REV = 0
+else
+  $(warning Unknown VENDOR_REV '$(VENDOR_REV)' for MACHINE '$(MACHINE)')
+  $(error Unknown VENDOR_REV)
+endif
+
+UBOOT_MACHINE = AS4610_54
```
Comment (lakesh-sharma) [inline machine/accton/accton_as4610_54/machine.make]: Generic comment : If we do not specify UBOOT_MACHINE in machine.make, would uboot
infra is not build.
e.g there could be platforms, which might use other uboots. i.e. not the one build by onie.

Comment (cbrune) [inline machine/accton/accton_as4610_54/machine.make]: @lakesh-sharma -- that is an unlikely scenario as UBOOT contains ONIE specific patches.  ONIE cannot be ONIE without the ONIE u-boot patches.  What a platform might do, however, is add additional U-Boot patches on top of the ONIE u-boot patches.

**Comment by cbrune** [comment]
Vidya Awesomeness!!  All the patches are pushed to master:

- 17fbb7a arm: accton AS4610-54: ONIE support for accton as4610-54 board
- 2264de8 arm: iproc: ONIE Support for Broadcom IPROC architecture
- c70efed u-boot: Support for U-Boot version 2012.10 in ONIE core
- b96d548 busybox sys_eeprom: Support reading/writing the "eeprom" exported in sysfs file

**Pull request closed without merging**